src/graphql/data/util.test.tsx

```diff
@@ -34,3 +34,25 @@ describe('fromGraphQLChain', () => {
     }
   })
 })
+
+describe('isValidUrlChainParam', () => {
+  it('should return true for valid chain name', () => {
+    const validChainName = 'ethereum'
+    expect(getValidUrlChainName(validChainName)).toBe(Chain.Ethereum)
+  })
+
+  it('should return false for undefined chain name', () => {
+    const undefinedChainName = undefined
+    expect(getValidUrlChainName(undefinedChainName)).toBe(undefined)
+  })
+
+  it('should return false for invalid chain name', () => {
+    const invalidChainName = 'invalidchain'
+    expect(getValidUrlChainName(invalidChainName)).toBe(undefined)
+  })
+
+  it('should return false for a misconfigured chain name', () => {
+    const invalidChainName = 'eThErEuM'
+    expect(getValidUrlChainName(invalidChainName)).toBe(undefined)
+  })
```

Comment:
maybe we should not be case sensitive?

Reply:
Valid point, might be worth discussing with product. Right now for TDP we're pretty forgiving, like this will resolve which I think is a little too far https://app.corn-staging.com/tokens/eTh%20Er%20%20%20%20____%20%20%20%20EuM/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48